datasets/prospective_memory.py

```diff
@@ -19,29 +28,69 @@ def cites_quote(quote: str, message: str) -> bool:
 
 
 @dataclass
-class ProspectiveMemoryDataset(GPTGenerated):
+class ProspectiveMemoryDataset(DatasetInterface):
     name: str = "Prospective Memory"
-    description: str = "Give the agent a fictitious quote, then ask it to append to the nth reply"
-    generation_file: Path = DATA_DIR.joinpath("gpt_generation_prompts/2-1_prospective_memory_test.json")
-    temperature: float = 0.5
-    uses_callback: bool = True
-    generation_model: str = GPT_4_TURBO_BEST
+    description: str = "Give the agent a quote, then ask it to append to the nth reply"
+    reset_message: str = "Forget my instruction to append a quote to one of your replies."
+
+    def request_num_generator(self) -> Iterator[int]:
+        indices = list(range(len(QUOTES)))
+        warned = False
+        while True:
+            self.random.shuffle(indices)
+            for i in indices:
+                yield i
+            if warned:
+                continue
+            warned = True
+            colour_print("red",
+                         "WARNING: The number of Quotes has been exceeded. There will inevitably "
+                         "be some repetitions.")
 
     def generate_examples(self, num_examples) -> List[TestExample]:
-        num_pattern = r"\d+(?:th|st|nd|rd)"
-        examples = super().generate_examples(num_examples)
-        for example in examples:
-            n = ordinal(self.random.randint(2, 9))
-            example.script[1] = re.sub(num_pattern, n, example.script[1])
-            example.script[1] += " Count your response to this message as the first response."
-            example.expected_responses[0] = re.sub(num_pattern, n, example.expected_responses[0])
+        examples = []
+
+        for _, request_num in zip(range(num_examples), self.request_num_generator()):
+            script = []
+            # Pick a quote
+            quote, author = QUOTES[request_num]
+            target = self.random.randint(2, 8)
+
+            n = ordinal(target)
+
+            script.append(f"'{quote}' - {author}")
```

Comment:
Suggestion: `repr(quote)` will add the quotation marks, and make them single or double depending on the content of the quote.